### ISSUE TITLE
feature: push to open-vsx.org

### ISSUE THREAD

**Issue by lrstanley** [open-policy-agent/vscode-opa#74]
Currently, it looks like this extension is only pushed to the VSCode Marketplace, however this means that wrapped/custom vscode implementations are unable to utilize this extension without manually downloading and installing it. It would be nice if it could get added to https://open-vsx.org/, to allow things like https://github.com/coder/code-server to use it.

**Comment by anderseknert** [comment]
Oh, nice! Yeah, that definitely sounds like a good idea 👍

**Comment by glisha** [comment]
I don't see any CI scripts in the repo here, so I guess publishing to the MS marketplace is done manually?

If so and based on the frequency of new versions, the same can be done with the  https://open-vsx.org/ so that the extension shows up in [code-server extensions tab.](https://github.com/coder/code-server/blob/main/docs/FAQ.md#why-cant-code-server-use-microsofts-extension-marketplace)

It would be nice for @tsandall  or someone that he is comfortable with to do it so the same namespace can be in both.

The steps are:

1. Sign the Eclipse Foundation Open VSX Publisher Agreement - https://www.eclipse.org/legal/open-vsx-registry-faq/#faq-6
2. Publish the packages - https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension

There is also a Github Actions ready script that can be configured to publish to both marketplaces at the same time. https://github.com/open-vsx/publish-extensions

**Comment by ChristianKuehnel** [comment]
Hey @tsandall, let me know if you would be fine with automatic releases to both marketplaces and I'd offer my help in setting this up using the scripts linked in the preivous post.

**Comment by anderseknert** [comment]
@ChristianKuehnel I've signed the various agreements and I'm looking into having it published. I'm not sure I understand namespaces though. Are we restricted to use the same namespace as the `publisher` field in the package.json file? That'd be "tsandall", but while Torin is still around (🙂), his handle is used as the publisher only because the VS Code marketplace won't allow it to be changed. Can we use `open-policy-agent` as our namespace when we publish this to Open VSX?

**Comment by hakan-77** [comment]
Hi all, is there any progress on publishing to the open-vsx marketplace? To give perspective, a popular VSCode package like Cline is installed ~1.2M times on Microsoft marketplace, and ~600K in the open one. Tools like Code-Server is gaining in popularity every day.

FYI - @anderseknert @tsandall @ChristianKuehnel

**Comment by anderseknert** [comment]
No, my question above remains unanswered, and since we haven't heard much from users asking for this, it's not been prioritized. So much to do, so little time, etc 😅

**Comment by hakan-77** [comment]
@anderseknert I know some packages that use different publisher tags for the open extension marketplace. But maybe this could also be an opportunity to rename and deploy a new one for Microsoft store too? The sooner the better :)

You could then mark the original extension deprecated and we would all know what to download. To publish to both marketplaces in one go using github Actions script: https://github.com/open-vsx/publish-extensions 

I hear you about other users not asking, but I hope the fact that one marketplace is proprietary, and the other is an open ecosystem can influence your decision..

**Comment by charlieegan3** [comment]
> But maybe this could also be an opportunity to rename and deploy a new one for Microsoft store too?

I agree that this would be ideal in the long run. But, it'd be nice to be able to get the extension out with the correct owner name (open-policy-agent) in the Open VSX marketplace without that needing to happen first.

Perhaps is there a way we can write the publish scripts such that a different or patched package.json is used?

**Comment by anderseknert** [comment]
> I know some packages that use different publisher tags for the open extension marketplace.

Can you help point as some? Would help if we knew better how to accomplish that.

I don’t think renaming here will happen, as it would mean starting from scratch with the rating, downloads and whatnot. While they’re arguably vanity metrics, they do matter to some for credibility, and people’s tendency to install.

**Comment by hakan-77** [comment]
Spring Boot Extension Pack was from Pivotal before it was acquired by VMWare on one marketplace vs. the other. 

I guess you either have to keep the namespace "tsandall" on both marketplaces, or write a simple script that replaces the namespace in package.json before publishing on the open marketplace. Your code and all the other artifacts will stay the same.

**Comment by pkoch** [comment]
Would love to have this so I can install this on Cursor and other Codiums!

Seems like this is growing a bit stale. What are the next steps that need to be taken here? If it's only about namespacing, I'd kindly ask that we not expand the scope of this ticket, and just publish with whatever we already have. That'll also make things more portable and less surprising for people using both VS Code and Codiums. Seeing a package that is called the same and looks the same inspired confidence.

For example, I don't want to look into https://marketplace.cursorapi.com/items/?itemName=glebbash.opa-highlight-only, which is what's on Open VSX today. How much of a fork is it? For how long will this maintainer not be compromised? etc.

**Comment by anderseknert** [comment]
The process is moving forward slowly, but it is moving.

**Comment by pkoch** [comment]
Do folks at home have any way to follow along?

**Comment by anderseknert** [comment]
We’ll try to be better about keeping you posted here.

In the meantime, would it help some if we pushed proper releases from now on, with downloadable VSX artifacts attached? Clearly not the marketplace experience we want, but it would at least simplify installing the extension in VS Code derivatives until we have that.

**Comment by pkoch** [comment]
I think that's a great babybstep! Thank you! 🚀

Meanwhile, if there's something the community can do to lessen your burden, I feel like a lot of folks would step forward to help! 🫡

**Comment by anderseknert** [comment]
Sorry for the delay, but at last we got a [new release](https://github.com/open-policy-agent/vscode-opa/releases/tag/v0.23.0) out to accompany Regal [v0.40.0](https://github.com/open-policy-agent/regal/releases). Hopefully making it a little easier for people to use in VS Code derivates in the meantime.

**Comment by manpreetsnehra** [comment]
On Vscodium, The released package gets installed and then gets auto uninstalled..

**Comment by avacaru** [comment]
Same here, tried to install the VSIX and got this warning: "This extension was reported to be problematic."

<img width="1267" height="592" alt="Image" src="https://github.com/user-attachments/assets/f953b78b-f7e2-4454-b228-29fb3fac8edf" />

After doing a `Developer: Reload Window`, I see this notification bottom right: "The extension 'Open Policy Agent' was found to be problematic and has been uninstalled"

<img width="490" height="122" alt="Image" src="https://github.com/user-attachments/assets/01029706-7632-43d2-81c1-c1264faf1475" />
